lms/services/user.py

```diff
@@ -61,28 +63,29 @@ def get(self, application_instance, user_id: str) -> User:
         :raises UserNotFound: if the User is not present in the DB
         """
         try:
-            existing_user = (
-                self._db.query(User)
-                .filter_by(
-                    application_instance=application_instance,
-                    user_id=user_id,
-                )
-                .one()
-            )
+            existing_user = self._user_search_query(
+                application_instance_id=application_instance.id, user_id=user_id
+            ).one()
+
         except NoResultFound as err:
             raise UserNotFound() from err
 
         return existing_user
 
-    def _find_existing_user(self, model_user: User) -> Optional[User]:
-        return (
-            self._db.query(User)
-            .filter_by(
-                application_instance_id=model_user.application_instance_id,
-                user_id=model_user.user_id,
-            )
-            .one_or_none()
-        )
+    def _user_search_query(self, application_instance_id, user_id) -> Query:
+        """Generate a query for searching for users."""
+
+        query = self._db.query(User)
```

Comment:
So in non-legacy sqlalchemy style I _think_ this would be something like:

```python
query = select(User)
```

Then when it comes time to actually execute the query you have to do:

```python
result = self._db.execute(query)
```